frontend/src/utils/Tabs.svelte

```diff
@@ -0,0 +1,81 @@
+<style>
+  .content {
+    position: sticky;
+    top: 0;
+    height: 100%;
+    width: 100%;
+    overflow: hidden;
+  }
+
+  button {
+    margin-bottom: 0;
+    border: 1px solid white;
+  }
+
+  .tabs button:focus {
+    border-bottom: 2px solid var(--main-bg-color);
+    outline: 0;
+  }
+
+  .tabs button {
+    margin-right: 0.5em;
+    border-bottom: 0px;
+  }
+
+  .content hr {
+    display: block;
+    height: 1px;
+    border: 0;
+    border-top: 1px solid white;
+    margin-top: -1px;
+    padding: 0;
+  }
+</style>
+
+<script>
+  import { onMount } from "svelte";
+  export let tabs, initTabId;
+  let displayType;
+
+  onMount(async () => {
+    document.getElementById(initTabId).click();
+  });
+
+  function resetTab() {
+    if (!tabs.map((x) => x.id).includes(displayType)) {
+      displayType = initTabId;
+    }
+  }
+
+  $: resetTab(tabs, initTabId, displayType);
+</script>
+
+<div class="content">
+  <div class="tabs">
+    {#each tabs as tab}
+      {#if displayType == tab.id}
+        <button
+          style="border-bottom: 2px solid var(--main-bg-color)"
+          id="{tab.id}"
+          on:click="{(e) => {
+            displayType = tab.id;
+          }}">{tab.title}</button
+        >
+      {:else}
+        <button
+          id="{tab.id}"
+          on:click="{(e) => {
+            displayType = tab.id;
+          }}">{tab.title}</button
+        >
+      {/if}
```

Comment:
In my opinion, it's a little bit cleaner to do it this way.

```suggestion
      <button
          class:selected={displayType === tab.id}
          id="{tab.id}"
          on:click="{(e) => {
            displayType = tab.id;
          }}">{tab.title}</button
        >
```

And then up in the `<style>` tag add:

``` css
button.selected {
  border-bottom: 2px solid var(--main-bg-color);
}
```

Reply:
this is cool, didn't know you could do that